.github/s2n_osx.sh

```diff
@@ -14,25 +14,25 @@
 # permissions and limitations under the License.
 #
 set -eu
-source codebuild/bin/s2n_setup_env.sh
 
+export S2N_LIBCRYPTO=openssl-3.4
 export CTEST_OUTPUT_ON_FAILURE=1
-BREWINSTLLPATH=$(brew --prefix openssl@1.1)
-OPENSSL_1_1_1_INSTALL_DIR="${BREWINSTLLPATH:-"/usr/local/Cellar/openssl@1.1/1.1.1?"}"
+BREWINSTLLPATH=$(brew --prefix openssl@3)
+OPENSSL_3_INSTALL_DIR="${BREWINSTLLPATH:-"/opt/homebrew/Cellar/openssl@3"}"
```

Comment:
I wonder if this should just be OPENSSL_INSTALL_DIR or even LIBCRYPTO_INSTALL_DIR so we never have to update all the references again. But I don't think plan to test awslc here, and we've probably got a while before openssl-4 :P